src/rt/sections_elf_shared.d

```diff
@@ -723,6 +723,11 @@ void scanSegments(in ref dl_phdr_info info, DSO* pdso)
             assert(!pdso._tlsSize); // is unique per DSO
             pdso._tlsMod = info.dlpi_tls_modid;
             pdso._tlsSize = phdr.p_memsz;
+
+            // align to multiple of size_t to avoid misaligned scanning
+            // (size is subtracted from TCB address to get base of TLS)
+            immutable mask = size_t.sizeof - 1;
+            pdso._tlsSize = (pdso._tlsSize + mask) & ~mask;
```

Comment:
Ah sorry, I wasn't aware that the patch is only necessary for the LDC branch due to this optimization that is not in upstream: https://github.com/ldc-developers/druntime/blob/ldc/src/rt/sections_elf_shared.d#L996
I'll close this and resubmit against LDC's fork.